### PR TITLE
Issues/58053 - Set default textBaseline to alphabetic in the Table widget

### LINKED ISSUE #58053: Assertion failure when using baseline alignment in Table widget
# Steps to reproduce

Run the following app:

```dart
void main() {
  runApp(FailingApp());
}

class FailingApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'textBaseline assertion failure',
      home: Scaffold(
        body: Table(
          defaultVerticalAlignment: TableCellVerticalAlignment.baseline,
          children: <TableRow>[
            TableRow(
              children: <Widget>[
                Text('SCI - Overhead'),
                TextField(),
              ],
            ),
          ],
        ),
      ),
    );
  }
}
```

# Expected Behavior

You expect the table cells to be baseline aligned

# Actual behavior

```
flutter: ══╡ EXCEPTION CAUGHT BY RENDERING LIBRARY ╞═════════════════════════════════════════════════════════
flutter: The following assertion was thrown during performLayout():
flutter: 'package:flutter/src/rendering/table.dart': Failed assertion: line 1047 pos 22: 'textBaseline !=
flutter: null': is not true.
flutter: 
flutter: Either the assertion indicates an error in the framework itself, or we should provide substantially
flutter: more information in this error message to help you determine and fix the underlying cause.
flutter: In either case, please report this assertion by filing a bug on GitHub:
flutter:   https://github.com/flutter/flutter/issues/new?template=BUG.md
flutter: 
flutter: The relevant error-causing widget was:
flutter:   Table file:///Users/tvolkert/project/sci/apps/payouts/lib/main.dart:41:15
flutter: 
flutter: When the exception was thrown, this was the stack:
flutter: #2      RenderTable.performLayout (package:flutter/src/rendering/table.dart:1047:22)
flutter: #3      RenderObject.layout (package:flutter/src/rendering/object.dart:1767:7)
flutter: #4      MultiChildLayoutDelegate.layoutChild (package:flutter/src/rendering/custom_layout.dart:171:11)
flutter: #5      _ScaffoldLayout.performLayout (package:flutter/src/material/scaffold.dart:480:7)
flutter: #6      MultiChildLayoutDelegate._callPerformLayout (package:flutter/src/rendering/custom_layout.dart:240:7)
flutter: #7      RenderCustomMultiChildLayoutBox.performLayout (package:flutter/src/rendering/custom_layout.dart:399:14)
flutter: #8      RenderObject.layout (package:flutter/src/rendering/object.dart:1767:7)
flutter: #9      RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:113:13)
flutter: #10     RenderObject.layout (package:flutter/src/rendering/object.dart:1767:7)
flutter: #11     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:113:13)
flutter: #12     _RenderCustomClip.performLayout (package:flutter/src/rendering/proxy_box.dart:1267:11)
flutter: #13     RenderObject.layout (package:flutter/src/rendering/object.dart:1767:7)
flutter: #14     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:113:13)
flutter: #15     RenderObject.layout (package:flutter/src/rendering/object.dart:1767:7)
flutter: #16     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:113:13)
flutter: #17     RenderObject.layout (package:flutter/src/rendering/object.dart:1767:7)
flutter: #18     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:113:13)
flutter: #19     RenderObject.layout (package:flutter/src/rendering/object.dart:1767:7)
flutter: #20     RenderStack.performLayout (package:flutter/src/rendering/stack.dart:556:15)
flutter: #21     RenderObject.layout (package:flutter/src/rendering/object.dart:1767:7)
flutter: #22     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:113:13)
flutter: #23     RenderObject.layout (package:flutter/src/rendering/object.dart:1767:7)
flutter: #24     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:113:13)
flutter: #25     RenderObject.layout (package:flutter/src/rendering/object.dart:1767:7)
flutter: #26     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:113:13)
flutter: #27     RenderObject.layout (package:flutter/src/rendering/object.dart:1767:7)
flutter: #28     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:113:13)
flutter: #29     RenderObject.layout (package:flutter/src/rendering/object.dart:1767:7)
flutter: #30     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:113:13)
flutter: #31     RenderObject.layout (package:flutter/src/rendering/object.dart:1767:7)
flutter: #32     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:113:13)
flutter: #33     RenderOffstage.performLayout (package:flutter/src/rendering/proxy_box.dart:3219:13)
flutter: #34     RenderObject.layout (package:flutter/src/rendering/object.dart:1767:7)
flutter: #35     _RenderTheatre.performLayout (package:flutter/src/widgets/overlay.dart:700:15)
flutter: #36     RenderObject.layout (package:flutter/src/rendering/object.dart:1767:7)
flutter: #37     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:113:13)
flutter: #38     RenderObject.layout (package:flutter/src/rendering/object.dart:1767:7)
flutter: #39     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:113:13)
flutter: #40     RenderObject.layout (package:flutter/src/rendering/object.dart:1767:7)
flutter: #41     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:113:13)
flutter: #42     RenderObject.layout (package:flutter/src/rendering/object.dart:1767:7)
flutter: #43     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:113:13)
flutter: #44     RenderObject.layout (package:flutter/src/rendering/object.dart:1767:7)
flutter: #45     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:113:13)
flutter: #46     RenderObject.layout (package:flutter/src/rendering/object.dart:1767:7)
flutter: #47     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:113:13)
flutter: #48     RenderObject.layout (package:flutter/src/rendering/object.dart:1767:7)
flutter: #49     RenderView.performLayout (package:flutter/src/rendering/view.dart:167:13)
flutter: #50     RenderObject._layoutWithoutResize (package:flutter/src/rendering/object.dart:1630:7)
flutter: #51     PipelineOwner.flushLayout (package:flutter/src/rendering/object.dart:887:18)
flutter: #52     RendererBinding.drawFrame (package:flutter/src/rendering/binding.dart:402:19)
flutter: #53     WidgetsBinding.drawFrame (package:flutter/src/widgets/binding.dart:865:13)
flutter: #54     RendererBinding._handlePersistentFrameCallback (package:flutter/src/rendering/binding.dart:284:5)
flutter: #55     SchedulerBinding._invokeFrameCallback (package:flutter/src/scheduler/binding.dart:1113:15)
flutter: #56     SchedulerBinding.handleDrawFrame (package:flutter/src/scheduler/binding.dart:1052:9)
flutter: #57     SchedulerBinding.scheduleWarmUpFrame.<anonymous closure> (package:flutter/src/scheduler/binding.dart:861:7)
flutter: (elided 13 frames from class _AssertionError, class _RawReceivePortImpl, class _Timer, dart:async, and dart:async-patch)
flutter: 
flutter: The following RenderObject was being processed when the exception was fired: RenderTable#f4f8b relayoutBoundary=up1 NEEDS-LAYOUT NEEDS-PAINT NEEDS-COMPOSITING-BITS-UPDATE:
flutter:   creator: Table ← _BodyBuilder ← MediaQuery ← LayoutId-[<_ScaffoldSlot.body>] ←
flutter:     CustomMultiChildLayout ← AnimatedBuilder ← DefaultTextStyle ← AnimatedDefaultTextStyle ←
flutter:     _InkFeatures-[GlobalKey#6e62e ink renderer] ← NotificationListener<LayoutChangedNotification> ←
flutter:     PhysicalModel ← AnimatedPhysicalModel ← ⋯
flutter:   parentData: offset=Offset(0.0, 0.0); id=_ScaffoldSlot.body (can use size)
flutter:   constraints: BoxConstraints(0.0<=w<=1213.0, 0.0<=h<=684.0)
flutter:   size: MISSING
flutter:   default column width: FlexColumnWidth(1.0)
flutter:   table size: 2×1
flutter:   column offsets: 0.0, 606.5
flutter:   row offsets: 0.0
flutter: This RenderObject had the following descendants (showing up to depth 5):
flutter:     child (0, 0): RenderParagraph#73f52 NEEDS-LAYOUT NEEDS-PAINT
flutter:       text: TextSpan
flutter:     child (1, 0): RenderIgnorePointer#9ee2e NEEDS-LAYOUT NEEDS-PAINT NEEDS-COMPOSITING-BITS-UPDATE
flutter:       child: RenderMouseRegion#4c3d0 NEEDS-LAYOUT NEEDS-PAINT NEEDS-COMPOSITING-BITS-UPDATE
flutter:         child: RenderSemanticsAnnotations#c0070 NEEDS-LAYOUT NEEDS-PAINT NEEDS-COMPOSITING-BITS-UPDATE
flutter:           child: RenderPointerListener#ddf27 NEEDS-LAYOUT NEEDS-PAINT NEEDS-COMPOSITING-BITS-UPDATE
flutter:             child: _RenderDecoration#2c064 NEEDS-LAYOUT NEEDS-PAINT NEEDS-COMPOSITING-BITS-UPDATE
```

### PR DESCRIPTION
**Pull request by ramnavan**
## Description

`textBaseline` when not explicitly set causes issues when `defaultVerticalAlignment` is set to `TableCellVerticalAlignment.baseline`. As discussed [here](https://github.com/flutter/flutter/issues/58053), the solution is to set default value to textBaseline.

## Related Issues

Closes [58053](https://github.com/flutter/flutter/issues/58053)

## Tests

I added the following test:

`Table widget - Default textBaseline is set to TableBaseline.alphabetic` test has been added to `table_test.dart`

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- `[x]` I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- `[x]` I signed the [CLA].
- `[x]` I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- `[x]` I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- `[x]` I updated/added relevant documentation (doc comments with `///`).
- `[x]` All existing and new tests are passing.
- `[x]` The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- `[x]` I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- `[x]` No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*
   - [ ] I wrote a design doc: https://flutter.dev/go/template *Replace this with a link to your design doc's short link*
   - [ ] I got input from the developer relations team, specifically from: *Replace with the names of who gave advice*
   - [ ] I wrote a migration guide: https://flutter.dev/go/breaking-changes-template *Replace this with a link to a pull request that adds the migration guide to https://flutter.dev/docs/release/breaking-changes*

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: packages/flutter/test/widgets/table_test.dart [hunk 1]
```diff
@@ -932,5 +932,28 @@ void main() {
     },
   );
 
+  testWidgets(
+    'Table widget - Default textBaseline is set to TableBaseline.alphabetic',
+    (WidgetTester tester) async {
+      await tester.pumpWidget(
```
Comment (chunhtai) [inline packages/flutter/test/widgets/table_test.dart:938]: indent seems to be off

**Review by chunhtai — APPROVED** [review]
LGTM, except for the minor nit

File: packages/flutter/test/widgets/table_test.dart (hunk 1 above)
Comment (ramnavan) [reply packages/flutter/test/widgets/table_test.dart:938]: Thanks for the review. I addressed the indent and other minor styling issues (like adding trailing `,`) and pushed.

**Comment by chunhtai** [comment]
analyzer is complaining about version mismatch, you can fix that by rebasing onto master

**Comment by ramnavan** [comment]
Thanks @chunhtai . Rebased and pushed. Looks like all checks are passing but flutter-build is still broken

**Comment by Hixie** [comment]
We should revert this. We have no way to know that the correct baseline is alphabetic.

**Pull request merged**